src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
I wasn't sure if you wanted to configure the two separately. `Application` is used for loading the entry assembly. `EnvironmentApplication` is used to set the `ApplicationName` of the `IApplicationEnvironment` service.